ngapp/src/app/admin-components/find-user/find-user.component.ts

```diff
@@ -49,6 +51,18 @@ export class FindUserComponent implements OnInit {
   }
   dataLoaded: boolean = true;
 
+  validSearchText() {
+    try {
+      new RegExp(this.searchText, 'i'); // i for case insensitive
+    }
+    catch (e) {
+      console.log('Caught error:', e);
+      alert("invalid regular expression");
+      return false;
+    }
+    return true;
+  }
+
   searchUsers() {
     this.dataLoaded = false;
     this.userResults = [];
```

Comment:
This looks great -- nice catch!

Maybe we could put the valid regex check inside `searchUsers()` -- this function also gets called by the 'search' button in the html template, so saves us an `if` statement.

```suggestion
    if (!validTextSearch()) return;
    this.dataLoaded = false;
    this.userResults = [];
```